pkgs/development/python-modules/fastexcel/default.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  rustPlatform,
+
+  # nativeBuildInputs
+  cargo,
+  rustc,
+
+  # dependencies
+  pyarrow,
+  typing-extensions,
+
+  # tests
+  pandas,
+  polars,
+}:
+
+buildPythonPackage rec {
+  pname = "fastexcel";
+  version = "0.12.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-XcLx/z3f9skz0fkkrdf39dwgOIwecLIq+uDyizxPAck=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit pname version src;
+    hash = "sha256-JGDNqRF264hNAjQ9bwJnBsQgAcqJjreEbgRZAA58JnY=";
+  };
+
+  nativeBuildInputs = [
+    cargo
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+    rustc
+  ];
+
+  dependencies = [
+    pyarrow
+    typing-extensions
+  ];
+
+  optional-dependencies = {
+    pandas = [
+      pandas
+    ];
+    polars = [
+      polars
+    ];
+  };
+
+  pythonImportsCheck = [
+    "fastexcel"
+  ];
+
+  # No tests present in the pypi archive
+  doCheck = false;
```

Comment:
Then we need to fetch from git https://github.com/ToucanToco/fastexcel/tree/main/python/tests

Reply:
Did you try setting sourceRoot?